gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -69,6 +68,13 @@ def update_buildfiles_for_dependency(buildfiles:, dependency:)
 
           buildfile = files.find { |f| f.name == new_req.fetch(:file) }
 
+          # Exception raised to handle issue that arises when buildfiles function (see this file)
+          # removes the build file that contains the dependency itself. So no build file exists to
+          # update dependency, This behaviour is evident for extremely small number of users
+          # that have added separate repos as sub-modules in parent projects
+
```

Comment:
I found this comment a little confusing, so can we reword it to provide better context to the reader?

I took a stab at it, but I'm not 100% sure I fully understand the issue, so if I misstated something, please correct my wording

> Currently, Dependabot assumes that Gradle projects using Gradle submodules are all in a single repo. However, some projects are actually using git submodule references for the Gradle submodules. When this happens, Dependabot's FileFetcher thinks the Gradle submodules are eligible for update, but then the FileUpdater filters out the git submodule reference from the build file. So we end up with no relevant build file, leaving us with no way to update that dependency.
> TODO: Figure out a way to actually navigate this rather than throwing an exception.